sunpy/net/fido_factory.py

```diff
@@ -77,6 +77,14 @@ def __getitem__(self, aslice):
         if isinstance(aslice, slice):
             ret = self._list[aslice]
 
+        # using the client's name for indexing the responses.
+        elif isinstance(aslice, str):
+            ret = []
+            for res in self._list:
+                clientname = res.client.__class__.__name__
+                if aslice.lower() == clientname.lower().split('client')[0]:
```

Comment:
We really need to make sure we document this somewhere so people know how to name their classes.

Reply:
I added this to a tood in https://github.com/sunpy/sunpy/pull/4387